lib/gpio/pin.toit

```diff
@@ -189,16 +189,18 @@ class Pin:
   wait_for value -> none:
     if get == value: return
     gpio_config_interrupt_ num true
+    expected_state := value == 1 ? GPIO_STATE_UP_ : GPIO_STATE_DOWN_
     try:
       // Make sure the pin didn't change to the expected value while we
       // were setting up the interrupt.
-      if get == value: return
-      expected_state := value == 1 ? GPIO_STATE_UP_ : GPIO_STATE_DOWN_
-      state := state_.wait_for_state expected_state
-      state_.clear_state expected_state
+      // The loop is to prevent accidental wakeups.
```

Comment:
So how does the wakeup happen? I feel that I need to understand this better. With this change, it feels like you might drop signals (say the pin goes high, so we wakeup, but then it is low again before we ask for the value).

Reply:
Rewritten.